itunes_rss.go

```diff
@@ -0,0 +1,196 @@
+package feeds
+
+// itunes rss support
+// validation done according to spec here:
+//    https://help.apple.com/itc/podcasts_connect/#/itcb54353390
+
+import (
+	"encoding/xml"
+	"fmt"
+	"time"
+)
+
+// private wrapper around the RssFeed which gives us the <rss>..</rss> xml
+type ItunesRssFeedXml struct {
+	XMLName          xml.Name `xml:"rss"`
+	Version          string   `xml:"version,attr"`
+	ContentNamespace string   `xml:"xmlns:content,attr"`
+	Channel          *ItunesRssFeed
+	ItunesNamespace  string `xml:"xmlns:itunes,attr"`
+}
+
+type ItunesRssFeed struct {
+	XMLName        xml.Name `xml:"channel"`
+	Title          string   `xml:"title"`       // required
+	Link           string   `xml:"link"`        // required
+	Description    string   `xml:"description"` // required
+	Language       string   `xml:"language,omitempty"`
+	Copyright      string   `xml:"copyright,omitempty"`
+	ManagingEditor string   `xml:"author,omitempty"` // Author used
+	WebMaster      string   `xml:"webMaster,omitempty"`
+	PubDate        string   `xml:"pubDate,omitempty"`       // created or updated
+	LastBuildDate  string   `xml:"lastBuildDate,omitempty"` // updated used
+	Category       string   `xml:"category,omitempty"`
+	Generator      string   `xml:"generator,omitempty"`
+	Docs           string   `xml:"docs,omitempty"`
+	Cloud          string   `xml:"cloud,omitempty"`
+	Ttl            int      `xml:"ttl,omitempty"`
+	Rating         string   `xml:"rating,omitempty"`
+	SkipHours      string   `xml:"skipHours,omitempty"`
+	SkipDays       string   `xml:"skipDays,omitempty"`
+	Image          *RssImage
+	TextInput      *RssTextInput
+	IImage         *ItunesImage `xml:"itunes:image"`
+	ICategory      *ItunesCategory
+	IExplicit      string `xml:"itunes:explicit"`
+	IAuthor        string `xml:"itunes:author,omitempty"`
+	IOwner         *ItunesOwner
+	IType          string           `xml:"itunes:type,omitempty"`
+	INewFeedUrl    string           `xml:"itunes:new-feed-url,omitempty"`
+	IBlock         string           `xml:"itunes:block,omitempty"`
+	IComplete      string           `xml:"itunes:complete,omitempty"`
+	Items          []*ItunesRssItem `xml:"item,omitempty"`
+}
+
+type ItunesImage struct {
+	Href string `xml:"href,attr"`
+}
+
+type ItunesCategory struct {
```

Comment:
Public comment needs to match godoc conventions as above  - all of these need comments if exported